src/reason_pprint_ast.ml

```diff
@@ -3014,6 +3011,23 @@ class printer  ()= object(self:'self)
         in
         let rightItm = self#simplifyUnparseExpr rightExpr in
         Some (label ~space:forceSpace (atom prefixStr) rightItm)
+      | (Infix s, [(_, e1); (_, e2)]) when s.[0] = '#' ->
```

Comment:
Pretty printing using just the `rules` array with `##` being an `Infix` op does this:
```
bla #= 10;

(bla #= Some) 10;

(bla #= someFunc) (Some 10);

(test ## var #= Some) 10;

obj ##. prop;

obj ##. prod := exp;

(preview ## style ## border #= Js.string) "1px black dashed";

x ## y ## z #= (xxxx ## yyyy) ## zzzz;

let result = js_method_run1 (!react)#createElement foo;

let foo a => a;

id #= foo 2;

add (zz ## yy) (xx ## ww);

/* These should print the same */
let res = x ## y + z ## q; /* AST */

let res = x ## y + z ## q; /* Minimum parens */

/* These should print the same */
let res = y + z ## q ## a; /* AST */

let res = y + z ## q ## a; /* Min parens */

/* Make sure it's actually parsed as left precedence
 * and that is maintained when printed */
let res = z ## (q ## a); /* AST */

let res = z ## (q ## a); /* Min parens */
```

This does the right thing for `##` but might need to be special cased for `#=` which has a special meaning. How many of those symbols starting with `#` have special meaning (i.e.: aren't expected to have the same semantics as `##`)?